airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/state/object_storage/ObjectStorageDestinationStateManager.kt

```diff
@@ -58,7 +58,7 @@ class ObjectStorageDestinationState(
 
         return client
             .list(prefix)
-            .filter { matcher.match(it.key) != null }
+            .filter { stream.includeFiles || matcher.match(it.key) != null }
```

Comment:
@johnny-schmidt because of the prefix matching we _shouldn't_ delete data from other streams, but I wonder if there's a pattern where the `prefix` overlaps somehow. Can you confirm that's not possible? I'm reading through the code now.

Reply:
Maybe this is a case of buyer-beware if configured to do something like that?